src/slogger/slogger.cc

```diff
@@ -44,6 +47,54 @@ static safs::log_level::LogLevel log_level_from_syslog(int priority) {
 	return kSyslogToLevel[std::min<int>(priority, kSyslogToLevel.size())];
 }
 
+std::expected<safs::log_level::LogLevel, std::string>
+safs::log_level_from_string(const std::string &level) {
+	const auto lower_level = boost::algorithm::to_lower_copy(level);
+	if (lower_level == "trace") {
+		return safs::log_level::trace;
+	}
+	if (lower_level == "debug") {
+		return safs::log_level::debug;
+	}
+	if (lower_level == "info") {
+		return safs::log_level::info;
+	}
+	if (lower_level == "warn" || lower_level == "warning") {
+		return safs::log_level::warn;
+	}
+	if (lower_level == "err" || lower_level == "error") {
+		return safs::log_level::err;
+	}
+	if (lower_level == "crit" || lower_level == "critical") {
+		return safs::log_level::critical;
+	}
+	if (lower_level == "off") {
+		return safs::log_level::off;
+	}
+	return std::unexpected<std::string>("Invalid log level: " + level);
+}
+
+std::string safs::log_level_to_string(safs::log_level::LogLevel level) {
+	using safs::log_level::LogLevel;
+	switch (level) {
+	case LogLevel::trace:
+		return "trace";
+	case LogLevel::debug:
+		return "debug";
+	case LogLevel::info:
+		return "info";
+	case LogLevel::warn:
+		return "warn";
+	case LogLevel::err:
+		return "err";
+	case LogLevel::critical:
+		return "crit";
+	case LogLevel::off:
+		break;
```

Comment:
Consider adding `default` clause to avoid possible future errors in case an integer is cast to LogLevel type

Reply:
I think in this case it should be an assert, as if it's anything, it's most likely a bug in the program